keps/sig-network/3766-referencegrant/README.md

```diff
@@ -0,0 +1,697 @@
+`authorization.k8s.io` API group, defined within the Kubernetes code base
+as part of the 1.27 release.
+
+We will take this opportunity to clarify and update the API after SIG-Auth
+feedback. This resource will start with v1alpha1 as the API version.
+
+
+### Risks and Mitigations
+
+#### No Default Implementation
+Similar to the Ingress and Gateway APIs, this API will be dependent on
+implementations by controllers that are not included by default in Kubernetes.
+This could lead to confusion for users. We'll need to rely heavily on
+documentation for this feature, tracking all uses of official Kubernetes APIs
+that support ReferenceGrant in a central place.
+
+#### Potential for Variations Among Implementations
+Because this relies on each individual controller to implement the logic,
+it is possible that implementations may become inconsistent. To avoid that,
+we'll provide a standard library for implementing ReferenceGrant. We'll
+also strongly recommend that every API that relies on ReferenceGrant 
+includes robust conformance tests covering this functionality. Existing
+Gateway API conformance tests can serve as a model for this.
+
+#### Cross-Namespace References may Weaken Namespace Boundaries
+Although we believe that the handshake required for cross-namespace references
+with ReferenceGrant ensures these references will be safe, it does potentially
+weaken existing namespace boundaries. We believe ReferenceGrant will have a net
+benefit on the ecosystem as it will allow workloads, secrets, and configuration
+to be deployed in separate namespaces that more clearly match up with desired
+authorization.
+
+## Design Details
+
+### General Notes
+
+#### `ReferenceGrant` is half of a handshake
+
+When thinking about ReferenceGrant, it is important to remember that it does not
+do anything by itself. It *Grants* the *possibility* of making a *Reference*
+across namespaces. It's intended that _another object_ (that is, the From object)
+complete the handshake by creating a reference to the referent object (the To
+object).
+
+#### `Resource` vs `Kind`
+
+When creating a metaresource (that is, a resource that targets other resources)
+like ReferenceGrant, it's important to consider if the metaresource uses the more
+common `Kind` or the more correct `Resource`.
+
+In the original Gateway API implementation, we chose to use `Kind` rather than
+`Resource`, mainly to improve the user experience. That is, it's easier users
+to take the value from the `kind` field at the top of the YAML they are already
+using, and put it straight into these fields, rather than needing to do a
+kind-resource lookup for every user's interaction with the API. @robscott even
+ended up making https://github.com/kubernetes/community/pull/5973 to clarify
+the API conventions.
+
+However, in discussion on this KEP, it's clear that the more generic nature of
+_this_ API requires the additional specificity that `Resource` provides.
```

Comment:
> Add, remove, or change fields in the ReferenceGrant API

&mdash; non-goals, above

We can pick either; maybe we shouldn't have both though.

I'm fine with the API change, and I think we should revise the non-goal to clarify

Reply:
Good call, removed from non-goals. We really were trying to avoid API changes, but it looks like we're going to be stuck with at least the kind -> resource one.